From: Simon McVittie <smcv@debian.org>
Date: Sun, 5 Mar 2023 16:16:12 +0000
Subject: keyboard: Update filter and sort when the display name changes

The display name in `InputWidget.name` can affect `sort_inputs()`
and `input_visible()`, so we should update it when we replace the
placeholder display name with the real one, and tell GTK to update its
state accordingly.

Signed-off-by: Simon McVittie <smcv@debian.org>
Bug-Debian: https://bugs.debian.org/1032382
Forwarded: https://gitlab.gnome.org/GNOME/gnome-initial-setup/-/issues/180
---
 gnome-initial-setup/pages/keyboard/cc-input-chooser.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/gnome-initial-setup/pages/keyboard/cc-input-chooser.c b/gnome-initial-setup/pages/keyboard/cc-input-chooser.c
index 3330517..2dfdc5a 100644
--- a/gnome-initial-setup/pages/keyboard/cc-input-chooser.c
+++ b/gnome-initial-setup/pages/keyboard/cc-input-chooser.c
@@ -582,8 +582,10 @@ update_ibus_active_sources (CcInputChooser *chooser)
         while (item) {
                 InputWidget *row;
                 GtkWidget *child;
+                GtkListBoxRow *list_box_row;
 
-                child = gtk_list_box_row_get_child (GTK_LIST_BOX_ROW (item));
+                list_box_row = GTK_LIST_BOX_ROW (item);
+                child = gtk_list_box_row_get_child (list_box_row);
 
 		row = get_input_widget (child);
                 item = gtk_widget_get_next_sibling (item);
@@ -600,7 +602,9 @@ update_ibus_active_sources (CcInputChooser *chooser)
                 if (engine_desc) {
                         name = engine_get_display_name (engine_desc);
                         gtk_label_set_text (GTK_LABEL (row->label), name);
-                        g_free (name);
+                        g_clear_pointer (&row->name, g_free);
+                        row->name = g_steal_pointer (&name);
+                        gtk_list_box_row_changed (list_box_row);
                 }
         }
 }
